src/workflows-app/SubmissionDetails.js

```diff
@@ -1,9 +1,471 @@
+      setRunsData(runsAnnotatedWithDurations);
+      setRunsFullyUpdated(runsResponse.fully_updated);
+    },
+    [signal, submissionId]
+  );
+
+  const loadRunsData = useCallback(
+    async (cbasProxyUrlDetails) => {
+      try {
+        if (cbasProxyUrlDetails.status !== AppProxyUrlStatus.Ready) {
+          const { cbasProxyUrlState } = await loadAppUrls(workspaceId, 'cbasProxyUrlState');
+
+          if (cbasProxyUrlState.status === AppProxyUrlStatus.Ready) {
+            await loadRuns(cbasProxyUrlState.status);
+          }
+        } else {
+          await loadRuns(cbasProxyUrlDetails.state);
+        }
+      } catch (error) {
+        notify('error', 'Error loading saved workflows', { detail: error instanceof Response ? await error.text() : error });
+      }
+    },
+    [loadRuns, workspaceId]
+  );
+
+  const loadRunSets = useCallback(
+    async (cbasUrlRoot) => {
+      try {
+        const runSets = await Ajax(signal).Cbas.runSets.get(cbasUrlRoot);
+        return runSets.run_sets;
+      } catch (error) {
+        notify('error', 'Error getting run set data', { detail: error instanceof Response ? await error.text() : error });
+      }
+    },
+    [signal]
+  );
+
+  const loadAllRunSets = useCallback(
+    async (cbasProxyUrlDetails) => {
+      try {
+        if (cbasProxyUrlDetails.status !== AppProxyUrlStatus.Ready) {
+          const { cbasProxyUrlState } = await loadAppUrls(workspaceId, 'cbasProxyUrlState');
+
+          if (cbasProxyUrlState.status === AppProxyUrlStatus.Ready) {
+            loadRunSets(cbasProxyUrlState.state).then((runSet) => {
+              if (runSet !== undefined) {
+                setRunSetData(runSet);
+                loadMethodsData(cbasProxyUrlDetails.state, runSet.method_id, runSet.method_version_id);
+              }
+            });
+          } else {
+            const cbasUrlState = cbasProxyUrlState.state;
+            const errorDetails = cbasUrlState instanceof Response ? await cbasUrlState.text() : cbasUrlState;
+            const additionalDetails = errorDetails ? `Error details: ${JSON.stringify(errorDetails)}` : '';
+            notify('warn', 'Error loading Workflows app', {
+              detail: `Workflows app not found. Will retry in 30 seconds. ${additionalDetails}`,
+              timeout: CbasPollInterval - 1000,
+            });
+          }
+        } else {
+          loadRunSets(cbasProxyUrlDetails.state).then((runSet) => {
+            if (runSet !== undefined) {
+              setRunSetData(runSet);
+              loadMethodsData(cbasProxyUrlDetails.state, runSet.method_id, runSet.method_version_id);
+            }
+          });
+        }
+      } catch (error) {
+        notify('error', 'Error getting run set data', { detail: error instanceof Response ? await error.text() : error });
+      }
+    },
+    [workspaceId, loadRunSets, loadMethodsData]
+  );
+
+  // helper for auto-refresh
+  const refresh = withBusyState(setLoading, async () => {
+    try {
```

Comment:
I think even with the changes to `loadAllRunSets` we might still need to start the refresh after CBAS is in ready state. Otherwise this refresh and the `usePollingEffect` will both be making calls to Leo to get app status for CBAS.